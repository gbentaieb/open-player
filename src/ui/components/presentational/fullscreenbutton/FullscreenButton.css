.PlayPauseButton {
  position: relative;
  margin: auto;
}
