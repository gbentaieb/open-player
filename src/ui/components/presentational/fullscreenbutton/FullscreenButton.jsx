import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FullscreenIcon from '@material-ui/icons/Fullscreen';
import ExitFullscreenIcon from '@material-ui/icons/FullscreenExit';

import Button from '../button/Button';

import style from './FullscreenButton.css';

class FullscreenButton extends Component {
  static propTypes = {
    onClick: PropTypes.func.isRequired,
    isFullscreen: PropTypes.bool.isRequired,
  }

  static defaultProps = {
    /* Add default props */
  }

  render() {
    return (
      <div className={style.PlayPauseButton}>
        <Button
          onClick={this.props.onClick}
          Icon={this.props.isFullscreen ? ExitFullscreenIcon : FullscreenIcon}
        />
      </div>
    );
  }
}

export default FullscreenButton;
