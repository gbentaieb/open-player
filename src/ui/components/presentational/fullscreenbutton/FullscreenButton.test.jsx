import React from 'react';
import FullscreenButton from './FullscreenButton';

const defaultProps = {
  onClick: () => {},
  isFullscreen: true,
};

describe('PlayPauseButton snapshot', () => {
  test('PlayPauseButton > Snapshot', () => {
    const wrapper = shallow(<FullscreenButton {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
