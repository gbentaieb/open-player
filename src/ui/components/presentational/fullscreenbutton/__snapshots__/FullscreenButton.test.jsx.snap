// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayPauseButton snapshot PlayPauseButton > Snapshot 1`] = `
<div>
  <WithStyles(Button)
    Icon={[Function]}
    onClick={[Function]}
  />
</div>
`;
